src/nativeMain/kotlin/api/EclairClient.kt

```diff
@@ -58,4 +60,19 @@ class EclairClient(private val apiHost: String, private val apiPassword: String)
             else -> ApiError(statusCode.value, "api error")
         }
     }
+
+    @OptIn(InternalAPI::class)
+    override suspend fun connect(uri: String): Either<ApiError, String> {
+        return try {
+            val response: HttpResponse = httpClient.post("$apiHost/connect") {
+                setBody(uri)
```

Comment:
You'll probably need to modify the `ContentNegotiation` setting line 39: requests are using multipart form data, responses are using json.